run-with-macadam.sh

```diff
@@ -0,0 +1,348 @@
+    echo "VM image extracted to ${DISK_IMAGE_PATH}."
+}
+
+# Ensures macadam tool is available
+function ensure_macadam_exists() {
+    echo "--- Setting up macadam ---"
+    if command -v macadam &>/dev/null; then
+        echo "macadam is already available in PATH."
+        return
+    fi
+
+    if [ -f "/opt/macadam/macadam" ]; then
+        export PATH="/opt/macadam:$PATH"
+        echo "macadam found in /opt/macadam and added to PATH."
+        return
+    fi
+
+    echo "macadam not found, downloading..."
+
+    local version="latest"
+    local url="https://github.com/crc-org/macadam/releases/download/${version}/macadam-linux-amd64"
+
+    echo "Downloading from ${url}"
+    sudo mkdir -p /opt/macadam
+    if ! curl -fL -o /tmp/macadam "${url}"; then
+        die "Failed to download macadam."
+    fi
+    sudo mv /tmp/macadam /opt/macadam/macadam
+    sudo chmod +x /opt/macadam/macadam
+
+    export PATH="/opt/macadam:$PATH"
+    echo "macadam downloaded to /opt/macadam and added to PATH."
+}
+
+# start the VM
+function start_macadam_vm() {
+    echo "--- Creating VM ---"
+    macadam init \
+        "${DISK_IMAGE_PATH}" \
+        --disk-size 31 \
+        --memory 11264 \
+        --name crc-ng \
+        --username core \
+        --ssh-identity-path "${PRIV_KEY_PATH}" \
+        --cpus 6 \
+        --cloud-init "${USER_DATA_PATH}" \
+        --log-level debug
+
+    if ! macadam start crc-ng --log-level debug; then
+        echo "Machine didn't come up in time"
+    fi
+}
+
+# Exposes a port from the VM to the host
+function forward_port_gvproxy() {
+    echo "--- Exposing port ---"
+    local expose_req='{"local":"127.0.0.1:6443","remote":"192.168.127.2:6443","protocol":"tcp"}'
+    if ! curl -f \
+        --unix-socket "${GVPROXY_SOCKET_PATH}" \
+        -X POST \
+        -H 'Content-Type: application/json' \
+        -d "${expose_req}" \
+        http://gvproxy/services/forwarder/expose; then
+        die "Failed to forward API server port: ${expose_req}"
+    fi
+}
+
+# add dns entries for API server and console
+function add_api_server_dns() {
+    echo "--- Adding DNS entries for API Server ---"
+    local CRC_TESTING_ZONE='{
+    "Name": "crc.testing.",
+    "Records": [
+        {
+            "Name": "gateway",
+            "IP": "192.168.127.1"
+        },
+        {
+            "Name": "host",
+            "IP": "192.168.127.254"
+        },
+        {
+            "Name": "api",
+            "IP": "192.168.127.2"
+        },
+        {
+            "Name": "api-int",
+            "IP": "192.168.127.2"
+        },
+        {
+            "Regexp": "crc-(.*?)-master-0",
+            "IP": "192.168.126.11"
+        }
+    ]
+}'
+    local APPS_CRC_TESTING_ZONE='{"Name": "apps-crc.testing.","DefaultIP": "192.168.127.2"}'
+    local CONTAINERS_INTERNAL_ZONE='{
+    "Name": "containers.internal.",
+    "Records": [
+        {
+            "Name": "gateway",
+            "IP": "192.168.127.254"
+        }
```

Comment:
I think this was for podman to work but since now podman-machine is de-facto for podman related stuff so we might need to remove it?

Reply:
removed